### PR TITLE
import: remove first stage dialog and add folder pane

### LINKED ISSUE #8596: Improve or revert two-step import process
There have been quite a few discussions on the darktable IRC channel in the recent days about the new two-step import process, and the general opinion (with which I concur) seems to be that it is a worse solution than the "import files" and "import folder" buttons from previous versions. It has caused confusion and, especially when importing single files and entire folders, it is felt to be one click too many.

It has been improved recently, in that you can just press enter to Ok through the second dialog when importing a whole folder, but I don't think that helps much. I personally find it hard to do that without at least reviewing the contents of the second dialog, which slows my workflow.

I'm willing to be persuaded that the new way is better, or that it can be improved, but I thought it worth raising an issue so that the discussion can continue in the open.

### LINKED ISSUE #8744: confusing names in import
Hi 

I don't know if this is the right place but here I go.

I'm using 3.5.0~git1792.411f33fde-1

When importing pictures there used to be two options: image... and folder...

Now there are import-in-place... and copy and import... The tooltips are saying 'select a folder to import from' and 'select a folder to import from (shortcut: CTRL + SHIFT +I)'. I think that's a lot more confusing than 'image' and 'folder'. Maybe it can/should be renamed again?
Screenshot from 2021-04-21 17-10-06
Screenshot from 2021-04-21 17-10-40

### PR DESCRIPTION
**Pull request by phweyland**
attempt to fix #8596
should also fix #8744

still miss:
- default root folder
- nice verbose tooltip on root folder 

image

### CODE REVIEW

**Comment by elstoc** [comment]
Thanks for working on this - it's looking much improved from the current two-step process.

The first time I open the screen I get the following:

Screenshot_2021-04-20_07-44-35

and the following error is shown in my console:

```
(darktable:317058): GLib-GIO-CRITICAL **: 07:43:36.159: g_file_enumerator_next_file: assertion 'G_IS_FILE_ENUMERATOR (enumerator)' failed
```
A folder seems to be selected when I open it (there are files displayed in the right-hand pane) but nothing is displayed in the left-hand-pane. If I click on the button at the top-left to choose a new root folder, and then select my home directory, darktable basically hangs. I get a lot more of the above errors in my console and when the import-in-place dialog comes back up, it's basically unchanged in appearance, with the Gtk file selection dialog behind it (my system won't let me screenshot this). I have to kill darktable at this point because it is unresponsive.

Other observations/suggestions: 

- The default dialog size needs to be increased. It should probably be the same as the preferences window now
- The right-hand pane could probably be allocated a greater percentage of the screen width and ideally it should be possible to resize the left/right pane proportions with the mouse
- Now that we're just selecting a base folder using the Gtk file dialog, I think the fix I added to #8541 might make more sense here
- The "folder" button above the tree is clickable but it only (I assume - since I can't see a tree) reverses the sort order. I think it would be better if only the triangle to the right was a clickable button and "folder" was just a label (like the similar button in the top panel).
- If you choose a single file in the right-hand pane, it would be great if the Enter key still clicked the "import in place" button

**Comment by elstoc** [comment]
I think my errors are to do with the fact that my home directory has a lot of stuff in it, including some ill-advised mounts to one of my servers. Moving those mounts I still get some of those errors on my console but the file list is displayed correctly. I would suggest maybe an additional check box to ignore hidden files/folders might be useful to improve the speed of loading the tree.

**Comment by phweyland** [comment]
> and the following error is shown in my console:

This is because I haven't not yet set a default value. As you suggested the default images folder should be used but I still have to find the way to get it. Advises are welcome.

>     The "folder" button above the tree is clickable but it only (I assume - since I can't see a tree) reverses the sort order.

Correct, that's the standard behavior. I can remove the title "folder", only the arrow will remain. ?

@elstoc thanks for testing.

>  Moving those mounts I still get some of those errors on my console

I don't manage any specific case (apart the "folder" one). If you share these console errors it may be possible to improve that.

> Also, the sort order should be remembered from one import to the next.

Right.

> I would suggest maybe an additional check box to ignore hidden files/folders might be useful

Right. I don't know yet how to do that however.

> And maybe we should have a drop-down to allow the user to select from the last 10 or so selected root folders.
> Could we lazy-load the tree only as people expand it?

Surely great options too. But let's start with the basics...

**Comment by elstoc** [comment]
> As you suggested the default images folder should be used but I still have to find the way to get it. Advises are welcome.

See [here](https://developer.gnome.org/glib/stable/glib-Miscellaneous-Utility-Functions.html#GUserDirectory) for some standard directories that you could use.

> Correct, that's the standard behavior. I can remove the title "folder", only the arrow will remain. ?

I thought you were building that yourself (so could separate the label and the button) but I see now it's just the treeview column header. Not a big deal.

> If you share these console errors it may be possible to improve that

They're all repetitions of the same error (possibly some odd type of file/folder in my folder structure) `(darktable:1023): GLib-GIO-CRITICAL **: 10:31:52.197: g_file_enumerator_next_file: assertion 'G_IS_FILE_ENUMERATOR (enumerator)' failed`

> Right. I don't know yet how to do that however.

I think you can use `G_FILE_ATTRIBUTE_STANDARD_IS_HIDDEN` - do you just negate that? I'm not sure

**Comment by quovadit** [comment]
First of all: great improvement @phweyland  :tada:

> See [here](https://developer.gnome.org/glib/stable/glib-Miscellaneous-Utility-Functions.html#GUserDirectory) for some standard directories that you could use.

I would not use standard directories like home or pictures but ui_last/import_last_directory

**Comment by elstoc** [comment]
> I would not use standard directories like home or pictures but ui_last/import_last_directory

That's a good idea but we'll still need a fallback for new installs where this won't be set. Could we pop up the selector immediately (i.e. initially restore the 2-step process) if it's not set?

**Comment by AlicVB** [comment]
really great work @phweyland !
what a great addition and improvement !

>  I would suggest maybe an additional check box to ignore hidden files/folders might be useful to improve the speed of loading the tree when the directory structure is large.

if "lazy-load" is hard to implement for now (I perfectly understand that, no worry) maybe we could at least add a "safety guard" to not go too far in hierarchy (something like a max recursion value, eventually configurable with a hidden pref)
Because sooner or later, we will have someone who select it's root folder (or /media/) and don't understand why dt is suddenly unresponsive :)

**Comment by elstoc** [comment]
> we will have someone who select it's root folder (or /media/) and don't understand why dt is suddenly unresponsive :)

If we put a max recursion value though, there's the issue that someone can't _reach_ the folder they want though, so it might have to be fairly high.

**Comment by phweyland** [comment]
changes made:

- default root folder (pick up last directory if exists)
- save folder order
- doesn't show hidden folder in folder pane
- save pane position

I've made a try with an hidden parameter. Allowing the hidden folders in /home for example stucks dt. The show hidden folders option still remain in filechooser to select the root folder.

I've also tried to grab the focus on import button whenever an image is selected, but that did not work. It works only on the first display of the dialog. Not sure why (I suspect it works only when done asynchronously after the selection change. The selection change event occurs too early).

**Comment by Nilvus** [comment]
I just love that, even without testing. Import module became really awesome and simple that way and it seems to also fix hover effect issue that @elstoc tried to fix with #8517 and #8541. @elstoc: do you confirm? If yes, your pr could probably be closed.

Great work again. Thanks @phweyland

**Comment by elstoc** [comment]
> hover effect issue that @elstoc tried to fix with #8517 and #8541. @elstoc: do you confirm? If yes, your pr could probably be closed

Yes I'll close them when this PR is merged.

**Comment by phweyland** [comment]
The dt freeze on huge folders is really an issue.

> Could we lazy-load the tree only as people expand it?

I think this is the only viable solution. I'll make a try.

**Comment by elstoc** [comment]
> I've also tried to grab the focus on import button whenever an image is selected, but that did not work

Try handling the enter button from within the two treeviews instead.

**Comment by phweyland** [comment]
> Try handling the enter button from within the two treeviews instead

Sorry, I don't get this.

**Comment by TurboGit** [comment]
I started testing... Really really nice ! We are getting there after all your work !

Issues I found:

- When I have selected the eye to show all thumbs changing the directory should refresh the thumbs.
- The folder top left should be on its own line otherwise a long folder path is not fully displayed (as we have spaces on this window I'm proposing to place the selection of the folder in its own line taking the whole dialog width).

**Comment by elstoc** [comment]
> Sorry, I don't get this.

I mean handle the "key-press-event" on the two panes and emit the dialog response if Enter is pressed. See for example https://github.com/elstoc/darktable/commit/2ff46e97f7c552907b841824268803d22bb28d73

**Comment by phweyland** [comment]
> I mean handle the "key-press-event" on the two panes and emit the dialog response if Enter is pressed. See for example [elstoc@2ff46e9](https://github.com/elstoc/darktable/commit/2ff46e97f7c552907b841824268803d22bb28d73), which does this for the right-hand pane.

Great. I guess that highlighting the import button is not necessary, I'll remove it and handle enter key.

**Comment by elstoc** [comment]
You might even be able to apply it to the entire dialog. Enter always imports, Esc always closes.

**Comment by elstoc** [comment]
Yeah that's much faster. For some reason if I expand a folder without selecting one first, the first folder in the list is automatically selected, and the right-hand pane populated - not sure what's doing that.

**Comment by phweyland** [comment]
> For some reason if I expand a folder in the left-hand pane without selecting one first, the first folder in the list is automatically selected, and the right-hand pane populated - not sure what's doing that.

I've seen that too, but need to understand ... as well.

**Comment by phweyland** [comment]
> > For some reason if I expand a folder in the left-hand pane without selecting one first, the first folder in the list is automatically selected, and the right-hand pane populated - not sure what's doing that.
> 
> I've seen that too, but need to understand ... as well.

I don't see ...  the selection changes between the click on expander and the call to _row_expand routine. 
BTW tagging has the same defect.
A bit annoying but not a roadblock however. Except if somebody has an idea I let it as is.

**Comment by phweyland** [comment]
> A horizontal scrollbar might be useful on the left-hand pane, for very tree-ish directory trees

I think I have to choose between ellipsizing and horizontal scrollbar. What's the best ?

**Comment by elstoc** [comment]
> Except if somebody has an idea I let it as is

You can probably change it to always deselect when expanding by getting the current selection with `gtk_tree_view_get_selection` and then running `gtk_tree_selection_unselect_all` on the returned value.

Alternatively you could change it so that the thing being expanded is automatically selected when expanding it (using `gtk_tree_selection_select_path`).

> I think I have to choose between ellipsizing and horizontal scrollbar. What's the best ?

If we're trying to make it file-browser-like, I would say a scrollbar is better. I'm not sure I've ever seen ellipsizing on a file browser. If you change the default window size to 1100x750 in `darktableconfig.xml.in` (to match preferences) you can give plenty of width to the directory tree by default. IMO 600x600 is too small now we have the extra pane.

**Comment by phweyland** [comment]
> You can probably change it to always deselect when expanding by getting the current selection with `gtk_tree_view_get_selection` and then running `gtk_tree_selection_unselect_all` on the returned value.
>
> Alternatively you could change it so that the path being expanded is automatically selected when expanding it (using `gtk_tree_selection_select_path`).

I understand the points but that's changing the global behavior of selection/expansion to workaround a treeview bug which occurs only at the first dt start (when there is no previous selected folder). I prefer the bug :).

> I would say a scrollbar is better

Easy to make a try. I'll do it.

> IMO 600x600 is too small now we have the extra pane. 

I don't like the huge window, but you are right, with the new pane, 600 width is not enough.

About the nice verbose tooltip on root folder ? Any advice ?

**Comment by elstoc** [comment]
> which occurs only at the first dt start (when there is no previous selected folder)

It also occurs when you select a new root folder but it's not a major issue. Hopefully people will only need to do this once.

> I don't like the huge window, but you are right, with the new pane, 600 width is not enough.

ok I was just going for consistency (we've a couple of dialogs using that size now) but whatever you think looks reasonable - people can always resize.

> About the nice verbose tooltip on root folder ? Any advice ?

Difficult to know what to say. Here's a first try, but I can't say I'm entirely happy with it yet:

```
choose the root of the folder tree below
most people will only need to set this once
try to choose the folder that contains most/all of your photographs (in sub-folders) 
so that you don't need to change the root frequently
```

**Comment by phweyland** [comment]
> > which occurs only at the first dt start (when there is no previous selected folder)
> 
> It also occurs when you select a new root folder but it's not a major issue. Hopefully people will only need to do this once.

you're right...

> but I can't say I'm entirely happy with it yet:

Thanks. More verbose than what I would think of, but that's a starting point. Easy to improve afterwards.

**Comment by elstoc** [comment]
> More verbose than what I would think of

On reflection you can probably remove the second line

**Comment by phweyland** [comment]
> It also occurs when you select a new root folder but it's not a major issue. Hopefully people will only need to do this once.

As the bug is around selection change event, we could avoid this not using it, and use the click on row instead to save the new selection... I'll make a try.

**Comment by phweyland** [comment]
> > It also occurs when you select a new root folder but it's not a major issue. Hopefully people will only need to do this once.
> 
> As the bug is around selection change event, we could avoid this not using it, and use the click on row instead to save the new selection... I'll make a try.

Tried, but unable to make it work, sorry (the first issue being not to be able to know if the clicked occurred on the expander or on the cell).

**Comment by phweyland** [comment]
Updated the import buttons tooltips. Unsure however what to do with the links to doc  ("lighttable_panels.html#import_from_fs") ... probably needs to be updated.

**Comment by elstoc** [comment]
> Tried, but unable to make it work

Me too, though I think the selection is somehow occurring before the folder expansion. If you hover over the "folders" title, you can see that the first entry gets highlighted as selected. If you choose a folder that only has a few sub-folders, clicking on the bottom (empty) of the treeview also triggers selection of the first item.

**Comment by elstoc** [comment]
@phweyland [this](https://github.com/elstoc/darktable/commit/8083cfb59ae2fa302239318b6cd2f18e91d8cd4b) commit is not perfect but it at least stops the files pane from getting populated (the first folder is still selected).

Also I noticed that the displayed sub-folders aren't correct. In the following screenshot you can see that the sub-folders also contain the parent folder name:

Screenshot_2021-04-21_18-58-13

**Comment by phweyland** [comment]
> Only seems to happen for the second level of folders as far as I can see.

Good catch. Yes, that's the iterative level. I'll check that.

> [this](https://github.com/elstoc/darktable/commit/8083cfb59ae2fa302239318b6cd2f18e91d8cd4b) commit is not perfect' but

I'll look at this too (does an expand click select the row ? that what I want to avoid).

**Comment by phweyland** [comment]
Not yet perfect too but not too far (parasite selection).

However 12f2516 has introduced a regression. The previous selected folder (last directory) is not opened any more.

**Comment by elstoc** [comment]
> However 12f2516 has introduced a regression. The previous selected folder (last directory) is not opened any more.

Yes I see that. The problem is when the last selected folder is more than two deep. You probably need to break the "last" path down and navigate through the tree programmatically, expanding and building nodes as you go. A simple search won't do it.

Nice work on the parasitic selection though - that was annoying. I think we're nearly there.

**Comment by elstoc** [comment]
FYI I've asked on IRC for some Windows/Mac users to test this as well to make sure there aren't any big cross-platform problems.

**Comment by elstoc** [comment]
Thanks. I'll give them a test later.

**Review by elstoc — APPROVED** [review]

**Comment by phweyland** [comment]
> Quick test. If I select a root directory that doesn't have many subdirectories, and then click on the empty space at the bottom, the first folder still shows as selected, though the right-hand pane doesn't get populated:

... yes, as there is no routine triggered afterwards, we have nothing to cleanup the parasitic selection.
The only way is to launch some time out routine to check this after each click on the view. Boring...

**Comment by phweyland** [comment]
generalize the folder expander fix to whatever click which is not an intended selection.

And thanks a lot @elstoc for your review, tests and brilliant suggestions !

**Comment by elstoc** [comment]
Fixes it for me. @TurboGit I think this is ready for you to review now. It's been tested on Mac (albeit on a commit from this morning) but not Windows yet.

File: src/libs/import.c [hunk 1]
```diff
@@ -783,83 +814,298 @@ static void _resize_dialog(GtkWidget *widget, dt_lib_module_t* self)
...
+    g_free(fullname);
+  } while(gtk_tree_model_iter_next(model, &child));
+}
 
-  // general info
-#ifdef HAVE_GPHOTO2
-  if(d->import_case == DT_IMPORT_CAMERA)
+static void _paned_position_changed(GtkWidget *widget, dt_lib_module_t* self)
+{
+  gint position = gtk_paned_get_position(GTK_PANED(widget));
```
Comment (TurboGit) [inline src/libs/import.c:987]: const

**Review by TurboGit — APPROVED** [review]
All good, thanks !

**Comment by AxelG-DE** [comment]
Gentlemen,

first of all, thank you very much for the willingness and finally achievement! That is great work.

I hope with my scenario I do not have a corner case...

- I do have per each trip one folder and inside that folder subfolders per each day.
- that means the "trip-folder" itself does not contain any pics but subfolders

That scenario, even recursive is ticked, I cannot (anymore) import (in place), as the import butten remains grayed out. I would need to select one by one the daily subfolders, which is kinda unfortunate...

Screenshot_20210422_230740

**Comment by phweyland** [comment]
> That scenario, even recursive is ticked, I cannot (anymore) import (in place), as the import butten remains grayed out. I would need to select one by one the daily subfolders, which is kinda unfortunate...

... no file is displayed on the right pane. You have excluded jpeg. May be stupid question but do you have other type of images in these folders ? raw whatever...

**Comment by AxelG-DE** [comment]
the subfolders contain RAWs only

My usecase worked fine in the past, BTW :)

**Comment by phweyland** [comment]
Oops, no folder is selected on the left pane. 
That  maybe an issue but the root folder is not scanned for files, only for folders.
As, you should select a root folder one (or several) levels up, and select in the folder pane the folder to be scanned.

**Comment by AxelG-DE** [comment]
I tried and did not make it. Maybe I am too tired ?

**Comment by phweyland** [comment]
> I tried and did not make it. Maybe I am too tired ?

But ... does that work now ?

**Comment by AxelG-DE** [comment]
Yeah I somehow made it, but is a bit trickier than in the past and also once I am inside, the film roll is set to one of the subfolders and hence I see only 5 instead of 389 pics. still kinda weirdo, but sorry, need to sleep now...

**Comment by phweyland** [comment]
In fact the basic use case is to set root folder to your "pictures" folder.
For upgrade purpose I've used your last saved directory as new root folder. This is misleading and seems to be a mistake.

**Comment by elstoc** [comment]
Maybe better to leave it empty and put some sort of "please choose a base folder" prompt into the side panel. That first step of selecting a base folder is gonna be hard for a lot of people I think. We need to think about how to ease that.

**Comment by phweyland** [comment]
I don't see better option than `G_USER_DIRECTORY_PICTURES`...

**Comment by elstoc** [comment]
If it exists. You can probably use home if it doesn't

**Comment by AxelG-DE** [comment]
First of all, sorry, that I cannot support these days for testing and commenting, as we are in the hot phase of moving homem and renovating our new house. I looked into it, as I go the message from system, that the related issue got closed.

> In fact the basic use case is to set root folder to your "pictures" folder.

Actually what I am doing shouldn't be so weird, it is actually an option of RPD and in fact I know that I am by far not alone with this setup.

I show here again, how it looks:
```
gerber@localhost 20200824_Rostock $ pwd
/home/gerber/Bilder/2020/20200824_Rostock
```

```
gerber@localhost 20200824_Rostock $ ls -l
insgesamt 68
drwxr-xr-x 2 gerber gerber  4096 24. Aug 2020  2014-12-31
drwxr-xr-x 2 gerber gerber  4096 24. Aug 2020  2020-08-22
drwxr-xr-x 2 gerber gerber 16384 24. Aug 2020  2020-08-23
drwxr-xr-x 2 gerber gerber 12288 24. Aug 2020  2020-08-24
drwxr-xr-x 2 gerber gerber 20480 25. Aug 2020  2020-08-25
```
Here we see, the first folder is named 2014, because that camera I used on that day, the battery was dead and I did  not realize it.

After importing now, I get merely that folder (maybe the first one, in any case) placed in the film roll for "collect images" and I see merely 5 instead of 398 images

the content of `gerber@localhost 20200824_Rostock $ find ./`
is pasted here https://dpaste.com/BG7UCMLWH

> For upgrade purpose I've used your last saved directory as new root folder. This is misleading and seems to be a mistake.

That one I did not get it to full extend, I just woke up and need coffee first. :-D (but then need to work)

I'd be very happy, we get this working again, otherwise I'd be in a big mess :blush:

**Comment by elstoc** [comment]
This is a misunderstanding of how this new functionality is supposed to work. But it is a hard thing to explain concisely and it's a misunderstanding that will be common. As an example, assume your image folders have the following general structure:

```
/home/chris/pictures
  2020
    January
    February
    : 
  2021
    January
    : 
```
The idea with this module is that you would select "pictures" as your root folder, and then _you would never need to change that root folder again_. For future imports you would just select a new sub-folder within that root.

However, unless people read the tooltips or the manual a lot will not understand this "intuitively" as it's not really a very common usage. To make this easier, I suggest that the first folder in the tree should always be the root folder (i.e. the entire tree should be displayed as above) and that first node should be expanded by default. That way the root folder itself can be selected if desired.

**Comment by parafin** [comment]
IMHO it's a very big assumption, that "one root folder will be enough for everyone". That's quite specific workflow, even if it's the most common. I would suggest to think about having a list of "root folders".

**Comment by elstoc** [comment]
@parafin: That's just been merged

**Comment by AxelG-DE** [comment]
I imported the 2020 folder completely 
(which is actually not so intuitive,
- as you need to step up two levels(!) higher in the "open folder" 
- (back to ~/home to see ~/Bilder in order to select ~/Bilder/2020 in (folder list"), 
- in order to select that folder in the "folder list",  
- rather than being able to import it right away in "open folder" or at least as chosen in "open folder"
)

But now, how could I see all photos of one trip (see my setup above)? I cleared film roll, but merely can select now folders with raws (sorted by date) but not the related main folder.

**To me** this still looks like a **severe regression**.

**Comment by elstoc** [comment]
Have you updated to the latest master? #8758 makes it a bit less painful, but I'm still not convinced you're using an optimal workflow.

**Comment by Nilvus** [comment]
> I imported the 2020 folder completely
> (which is actually not so intuitive,
> 
>     * as you need to step up two levels(!) higher in the "open folder"
> 
>     * (back to ~/home to see ~/Bilder in order to select ~/Bilder/2020 in (folder list"),
> 
>     * in order to select that folder in the "folder list",
> 
>     * rather than being able to import it right away in "open folder" or at least as chosen in "open folder"
>       )
> 
> 
> But now, how could I see all photos of one trip (see my setup above)? I cleared film roll, but merely can select now folders with raws (sorted by date) but not the related main folder.
> 
> **To me** this still looks like a **severe regression**.

I'm also convinced that you're not using an optimal workflow. This new way is way better than before.

On what you describe, I just see 2 steps after opening import window:

- just click to folder icon in top left, then go to wanted folder.
- then select images

Or, depending on which folder are set (or already used):
- click on folder where you want and just expand (like before)
-
or even select on of your last 10 used folders with a recent add in master by clicking on path folder showed in top of the window

**Comment by AxelG-DE** [comment]
> @AxelG-DE Have you updated to the latest master? #8758 makes it a bit less painful, but I'm still not convinced you're using an optimal workflow.
> 
> Also I've added some notes to [#8761 (comment)](https://github.com/darktable-org/darktable/issues/8761#issuecomment-825924922) with a suggestion on an optimal workflow (after #8758 is merged)

Yes I updated this morning. 

I think my workflow is very common (and recommended by many photographers, at least in DE) , that you devide by year, event or jobcode and inside event per day (e.g. travels lasted for weeks by the time we travelled) 

Also that workflow I use since years (>80k pics) including my 
differential backup (rsnapshot, 5y of history). So would be very unfortunate suddenly I would be asked to adopt all that, where dt worked very well in that scenario up to now.

**Comment by elstoc** [comment]
> I think my workflow is very common (and recommended by many photographers, at least in DE) , that you devide by year, event or jobcode and inside event per day (e.g. travels lasted for weeks by the time we travelled)

We're talking about your workflow using the dialog not your photo organisation strategy. In your case, I suspect you need to select `/home/gerber/Bilder` as your root directory, and then you just need to select your import directory in the left-hand pane. You should (probably) never need to change your root directory again.

**Comment by Nilvus** [comment]
Oups, just write an answer too fast. Just tested again and yes, there's an issue (maybe 3) if we select by error (or misunderstanding) last children folder directly as root folder:

- no images shown (even if that's not the best way, we should show them)
- but worst, I don't find a way to revert back to parent folder by clicking on icon folder on top left then choose a new root folder.

That last issue is the worst and a severe one I think. The only drawback is to select (if we have one) another folder in combobox menu where 10 last one used are saved

**Comment by elstoc** [comment]
I'm not convinced you guys are using latest master. These should both have been fixed yesterday.

**Comment by phweyland** [comment]
Make sure you have the last master. 

Select /home/gerber/builder in root folder.
now you can navigate through years and trips in the folder tree in the left pane. 
Select the trip 20200824_Rostock.
Tick recursive folders to see all photos of the multiple day trip. 
Next trip, no need to select again the root folder, just select it on the left pane.

**Comment by AxelG-DE** [comment]
@elstoc like I said, this what you described, is what I tried, but actually had to go even one level up to /home/gerber, so in the next window I could see 2020 to select (now I am at construction side with phone only, but most severe, after import I cannot get all photos of one event shown in lighttable, because in film roll I only see single days (did not check if there is a folder-filter).

**Comment by elstoc** [comment]
Sorry we're going in circles here. Please raise an issue with some screenshots or a screen recording of your workflow, because I don't have any issues with this.

**Comment by Nilvus** [comment]
> I'm not convinced you guys are using latest master. These should both have been fixed yesterday.

Quite strange, I was sure I've had last master (or to be precise, last one with last @phweyland PR (and by checking commit, that was...). But updating just now to last master and I can confirmed 2 issues I've post are fixed. All good now. Sorry for the noise.

@AxelG-DE: just update to last master (compile it now) and test again: I can see images (RAW and/or JPG if related checkbox is not checked) even in last children folder now. This really works great now.

**Comment by AxelG-DE** [comment]
O. K.
I'll check tonight, now need to continue my civil work, means demolition :-)

**Comment by AxelG-DE** [comment]
Ladies and Gentlemen,

I have now again updated to latest master
3.5.0-1887-ge33eb1469

With my, IMHO not so weird picture organisation (~/Bilder/YYYY/Event/'folder per day'), I **cannot make it** (what was formerly easily possibe) to **see** the **pictures of one trip/event/job-code in lighttable**(!) as one set. 

I can see in the import navigator, as @phweyland said, but not in lighttable finally. Neither film roll, nor folder does help in the "collect images" module.

Sorry to say, **currently I still see** it that way, that we have a **big regression**

**Comment by elstoc** [comment]
Please raise an issue.

**Comment by AxelG-DE** [comment]
[sigh!] that gonna wait until late summer or autumn even and then 3.6 is out... [No way for me to do anything earlier. That forced-quit if our current house has priority A+++]

**Comment by elstoc** [comment]
You've spent as much time commenting on this thread as it takes to raise an issue, and only a small subset of people will see these comments.

**Comment by elstoc** [comment]
> I cannot make it (what was formerly easily possibe) to see the pictures of one trip/event/job-code in lighttable(!) **as one set**.

This seems to be a different issue than what we were discussing earlier, and I'm still really unclear as to exactly what you're trying to do and exactly what problem you're having. The main reason I want you to raise an issue is so that you go through the issue template methodically and clearly describe what you're doing. From the above comment it sounds like the issue you're having now is that the files _are imported_ but you cannot find a way to filter them appropriately in the collect module.

If so, that's a different issue and I'm not sure how it can be caused by this PR.

**Pull request merged**